core/src/main/java/org/infinispan/topology/CacheTopology.java

```diff
@@ -223,4 +249,42 @@ public Integer getId() {
          return Collections.<Class<? extends CacheTopology>>singleton(CacheTopology.class);
       }
    }
+
+   /**
+    * Phase of topology change during rebalance. Using four phases guarantees these properties:
+    *
+    * 1. T(x+1).writeCH contains all nodes from Tx.readCH (this is the requirement for ISPN-5021)
+    * 2. Tx.readCH and T(x+1).readCH has non-empty subset of nodes (that will allow no blocking for read commands
+    *    && reading only entries node owns according to readCH)
+    *
+    * Old entries should be wiped out only after coming to the {@link #STABLE} phase.
```

Comment:
However, if you drop old entries at the beginning of READ_NEW_WRITE_ALL phase, some writes will store data in DC, and you do not wipe these. Actually, when a node is in this phase, it does not need to execute the writes locally, but it has to send writes to all *other* nodes according to writeCH. We could optimize that out in `LocalizedCacheTopology` by removing self as a write owner, and then we could move the clear to the point when we receive RNWA.